service/src/main/java/com/myhome/security/filters/CommunityAuthorizationFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.myhome.security.filters;
+
+import com.myhome.domain.User;
+import com.myhome.services.CommunityService;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommunityAuthorizationFilter extends BasicAuthenticationFilter {
+    private final CommunityService communityService;
+    private final String uuidPattern = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}";
```

Comment:
`uuidPattern` and `addAmenityRequestPattern` can be private static final (remember to user capitals with underscores in name for private static final)